@chrisbreiding tried to add the test you mentioned, bumped into two issues which I need help on:
1) I cannot commit `unner-specs/cypress/e2e/errors/uncaught.cy.js` eslint precommit hook fails:
```
✖ eslint --fix:

/Users/geritol/work/cypress/system-tests/project-fixtures/runner-specs/cypress/e2e/errors/uncaught.cy.js
   5:5  error  'cy' is not defined  no-undef
   6:5  error  'cy' is not defined  no-undef
  10:5  error  'cy' is not defined  no-undef
  11:5  error  'cy' is not defined  no-undef
  15:5  error  'cy' is not defined  no-undef
  16:5  error  'cy' is not defined  no-undef
  20:5  error  'cy' is not defined  no-undef
  21:5  error  'cy' is not defined  no-undef
  22:5  error  'cy' is not defined  no-undef
  26:5  error  'cy' is not defined  no-undef
  27:5  error  'cy' is not defined  no-undef
  28:5  error  'cy' is not defined  no-undef
  35:5  error  'cy' is not defined  no-undef
  39:5  error  'cy' is not defined  no-undef
  40:5  error  'cy' is not defined  no-undef
  48:5  error  'cy' is not defined  no-undef
  61:5  error  'cy' is not defined  no-undef
  67:5  error  'cy' is not defined  no-undef
  78:5  error  'cy' is not defined  no-undef

✖ 19 problems (19 errors, 0 warnings)
```

2) Tests fails because it seems It is expected for the error to be printed in an other place too aside from being loggeg (after the `>`)
<img width="434" alt="kép" src="https://user-images.githubusercontent.com/5496681/206674497-5e04fce3-ed3a-435b-9b09-98402e58c1c0.png">

Not sure where the logic of that is, maybe we need to add this transformation to a higher level in the code?

--- a/packages/driver/cypress/component/spec.cy.js
+++ b/packages/driver/cypress/component/spec.cy.js
@@ -1,4 +1,4 @@
-import sinon from 'sinon'
+const { sinon } = Cypress
 
 describe('component testing', () => {
   /** @type {Cypress.Agent<sinon.SinonSpy>} */

--- a/packages/driver/cypress/component/spec.cy.js
+++ b/packages/driver/cypress/component/spec.cy.js
@@ -1,3 +1,5 @@
+import sinon from 'sinon'
+
 describe('component testing', () => {
   /** @type {Cypress.Agent<sinon.SinonSpy>} */
   let uncaughtExceptionStub
@@ -12,17 +14,40 @@ describe('component testing', () => {
     })
   })
 
+  beforeEach(() => {
+    uncaughtExceptionStub.resetHistory()
+    document.querySelector('[data-cy-root]').innerHTML = ''
+  })
+
   it('fails and shows an error', () => {
+    cy.spy(Cypress, 'log').log(false)
     const $el = document.createElement('button')
 
     $el.innerText = `Don't click it!`
     $el.addEventListener('click', () => {
-      throw Error('An error!')
+      throw new Error('An error!')
     })
 
     document.querySelector('[data-cy-root]').appendChild($el)
     cy.get('button').click().then(() => {
       expect(uncaughtExceptionStub).to.have.been.calledOnceWithExactly(null)
+      expect(Cypress.log).to.be.calledWithMatch(sinon.match({ 'message': `Error: An error!`, name: 'uncaught exception' }))
+    })
+  })
+
+  it('fails and shows when a promise rejects with a string', () => {
+    cy.spy(Cypress, 'log').log(false)
+    const $el = document.createElement('button')
+
+    $el.innerText = `Don't click it!`
+    $el.addEventListener('click', new Promise((_, reject) => {
+      reject('Promise rejected with a string!')
+    }))
+
+    document.querySelector('[data-cy-root]').appendChild($el)
+    cy.get('button').click().then(() => {
+      expect(uncaughtExceptionStub).to.have.been.calledOnceWithExactly(null)
+      expect(Cypress.log).to.be.calledWithMatch(sinon.match({ 'message': `Error: "Promise rejected with a string!"`, name: 'uncaught exception' }))
     })
   })
 })

--- a/packages/driver/cypress/e2e/cypress/error_utils.cy.ts
+++ b/packages/driver/cypress/e2e/cypress/error_utils.cy.ts
@@ -671,14 +671,24 @@ describe('driver/src/cypress/error_utils', () => {
       }
     })
 
-    it('calls Cypress.log with error type and message when error is instance of Error', () => {
+    it('calls Cypress.log with error name and message when error is instance of Error', () => {
       $errUtils.logError(cypressMock, 'error', new Error('Some error'))
       expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: Some error`))
     })
 
-    it('calls Cypress.log with error type and message when error a string', () => {
-      $errUtils.logError(cypressMock, 'error', 'Some error')
-      expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: \"Some error\"`))
+    it('calls Cypress.log with error name and message when error a string', () => {
+      $errUtils.logError(cypressMock, 'error', 'Some string error')
+      expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: \"Some string error\"`))
+    })
+
+    it('calls Cypress.log with default error name and provided message message when error is an object with a message', () => {
+      $errUtils.logError(cypressMock, 'error', { message: 'Some object error with message' })
+      expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: Some object error with message`))
+    })
+
+    it('calls Cypress.log with error name and message when error is an object', () => {
+      $errUtils.logError(cypressMock, 'error', { err: 'Error details' })
+      expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: {"err":"Error details"}`))
     })
   })
 })

--- a/packages/driver/cypress/e2e/cypress/error_utils.cy.ts
+++ b/packages/driver/cypress/e2e/cypress/error_utils.cy.ts
@@ -5,6 +5,7 @@ allowTsModuleStubbing()
 import $stackUtils from '@packages/driver/src/cypress/stack_utils'
 import $errUtils, { CypressError } from '@packages/driver/src/cypress/error_utils'
 import $errorMessages from '@packages/driver/src/cypress/error_messages'
+import sinon from 'sinon'
 
 describe('driver/src/cypress/error_utils', () => {
   context('.modifyErrMsg', () => {
@@ -90,7 +91,7 @@ describe('driver/src/cypress/error_utils', () => {
     })
 
     it('attaches onFail to the error when it is a function', () => {
-      const onFail = function () {}
+      const onFail = function () { }
       const fn = () => $errUtils.throwErr(new Error('foo'), { onFail })
 
       expect(fn).throw().and.satisfy((err) => {
@@ -561,7 +562,7 @@ describe('driver/src/cypress/error_utils', () => {
 
     it('does not error if no last log', () => {
       state.returns({
-        getLastLog: () => {},
+        getLastLog: () => { },
       })
 
       const result = $errUtils.createUncaughtException({
@@ -658,6 +659,26 @@ describe('driver/src/cypress/error_utils', () => {
       })
 
       expect(unsupportedPlugin).to.eq(null)
+    })
+  })
+
+  context('.logError', () => {
+    let cypressMock
+
+    beforeEach(() => {
+      cypressMock = {
+        log: cy.stub(),
+      }
+    })
+
+    it('calls Cypress.log with error type and message when error is instance of Error', () => {
+      $errUtils.logError(cypressMock, 'error', new Error('Some error'))
+      expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: Some error`))
+    })
+
+    it('calls Cypress.log with error type and message when error a string', () => {
+      $errUtils.logError(cypressMock, 'error', 'Some error')
+      expect(cypressMock.log).to.have.been.calledWithMatch(sinon.match.has('message', `Error: \"Some error\"`))
     })
   })
 })

--- a/packages/driver/cypress/e2e/cypress/error_utils.cy.ts
+++ b/packages/driver/cypress/e2e/cypress/error_utils.cy.ts
@@ -5,7 +5,8 @@ allowTsModuleStubbing()
 import $stackUtils from '@packages/driver/src/cypress/stack_utils'
 import $errUtils, { CypressError } from '@packages/driver/src/cypress/error_utils'
 import $errorMessages from '@packages/driver/src/cypress/error_messages'
-import sinon from 'sinon'
+
+const { sinon } = Cypress
 
 describe('driver/src/cypress/error_utils', () => {
   context('.modifyErrMsg', () => {

--- a/packages/driver/src/cypress/error_utils.ts
+++ b/packages/driver/src/cypress/error_utils.ts
@@ -549,9 +549,11 @@ const errorFromUncaughtEvent = (handlerType: HandlerType, event) => {
     errorFromProjectRejectionEvent(event)
 }
 
-const logError = (Cypress, handlerType: HandlerType, err, handled = false) => {
+const logError = (Cypress, handlerType: HandlerType, err: unknown, handled = false) => {
+  const error = toLoggableError(err)
+
   Cypress.log({
-    message: `${err.name}: ${err.message}`,
+    message: `${error.name}: ${error.message}`,
     name: 'uncaught exception',
     type: 'parent',
     // specifying the error causes the log to be red/failed
@@ -570,6 +572,25 @@ const logError = (Cypress, handlerType: HandlerType, err, handled = false) => {
       return consoleObj
     },
   })
+}
+
+const isLogabbleError = (error: unknown): error is Error => {
+  return (
+    typeof error === 'object' &&
+    error !== null &&
+    'message' in error &&
+    typeof (error as Record<string, unknown>).message === 'string'
+  )
+}
+
+const toLoggableError = (maybeError: unknown): Error => {
+  if (isLogabbleError(maybeError)) return maybeError
+
+  try {
+    return new Error(JSON.stringify(maybeError))
+  } catch {
+    return new Error(String(maybeError))
+  }
 }
 
 const getUnsupportedPlugin = (runnable) => {

--- a/packages/driver/src/cypress/error_utils.ts
+++ b/packages/driver/src/cypress/error_utils.ts
@@ -553,7 +553,7 @@ const logError = (Cypress, handlerType: HandlerType, err: unknown, handled = fal
   const error = toLoggableError(err)
 
   Cypress.log({
-    message: `${error.name}: ${error.message}`,
+    message: `${error.name || 'Error'}: ${error.message}`,
     name: 'uncaught exception',
     type: 'parent',
     // specifying the error causes the log to be red/failed
@@ -574,7 +574,9 @@ const logError = (Cypress, handlerType: HandlerType, err: unknown, handled = fal
   })
 }
 
-const isLogabbleError = (error: unknown): error is Error => {
+interface LoggableError { name?: string, message: string }
+
+const isLoggabbleError = (error: unknown): error is LoggableError => {
   return (
     typeof error === 'object' &&
     error !== null &&
@@ -583,8 +585,8 @@ const isLogabbleError = (error: unknown): error is Error => {
   )
 }
 
-const toLoggableError = (maybeError: unknown): Error => {
-  if (isLogabbleError(maybeError)) return maybeError
+const toLoggableError = (maybeError: unknown): LoggableError => {
+  if (isLoggabbleError(maybeError)) return maybeError
 
   try {
     return new Error(JSON.stringify(maybeError))

